### PR TITLE
feat(demodata): Notify on demodata success and failure. 

### LINKED ISSUE #17859: Demo Data notifications
On demo data bucket creation: 
- [x] success long notification: "Successfully created dashboard for website monitoring demo data" -> click through to that dashboard
- [x] error if could not add demodata bucket. 
- [x] error if could not create demodata dashboard. 

- [x] if demo data bucket has already been added disable corresponding dropdown item. 
- [x] If delete demodata bucket -> delete demodata dashboard ?  no
- [x] If delete demodata dashboard -> delete demodata bucket ? no

### PR DESCRIPTION
**Pull request by ebb-tide**
Closes #17859
This PR completes error handling for the demodata feature in the UI
I moved the logic fetching demodata buckets to the demodata dropdown itself. 
And had to refactor notifications to allow passing a link to a notification. 
Improved style of demodata dropdown

### CODE REVIEW

File: ui/src/buckets/components/BucketsTab.tsx [hunk 1]
```diff
@@ -125,12 +108,7 @@ class BucketsTab extends PureComponent<Props, State> {
     const rightHeaderItems = (
       <>
         <FeatureFlag name="demodata">
-          {demoDataBuckets.length > 0 && (
-            <DemoDataDropdown
-              buckets={demoDataBuckets}
-              getMembership={getDemoDataBucketMembership}
-            />
-          )}
+          <DemoDataDropdown />
```
Comment (hoorayimhelping) [inline ui/src/buckets/components/BucketsTab.tsx:111]: nice change, love that all that logic is abstracted behind this component :+1:

File: ui/src/cloud/actions/demodata.ts [hunk 2]
```diff
@@ -60,85 +64,72 @@ export const getDemoDataBuckets = () => async (
...
-      const deleteResp = await deleteDashboard({
-        dashboardID: ddDashboard.id,
-      })
-      if (deleteResp.status !== 204) {
-        throw new Error(deleteResp.data.message)
-      }
-    }
-  } catch (error) {
-    throw new Error(error)
+    console.error(error)
```
Comment (hoorayimhelping) [inline ui/src/cloud/actions/demodata.ts:132]: same suggestion here for removing this - not sure if this is valuable since we're reporting the error. but it might be and i may be a dunce

File: ui/src/cloud/apis/demodata.ts [hunk 3]
```diff
@@ -1,56 +1,49 @@
...
-    console.error(error)
-    throw error
+  const {data} = await AJAX({
+    method: 'GET',
+    url: `${baseURL}/buckets`,
+  })
+
+  // if sampledata endpoints are not available in a cluster
+  // gateway responds with a list of links where 'buckets' field is a string
+  const buckets = get(data, 'buckets', false)
```
Comment (hoorayimhelping) [inline ui/src/cloud/apis/demodata.ts:26]: nit: `null` fits this better conceptually than `false`. `null` generally means has 'no value' and applies a little bit more cleanly than `false` does to an array. they both accomplish the same thing, it's just a little bit more conceptually sane to assign `null` to a collection than `false`.

File: ui/src/cloud/apis/demodata.ts [hunk 4]
```diff
@@ -1,56 +1,49 @@
...
-    })
-
-    if (response.status === '200') {
-      // a failed or successful membership POST to sampledata should return 204
-      throw new Error('Could not reach demodata endpoint')
-    }
-  } catch (error) {
-    console.error(error)
-    throw error
+  const response = await AJAX({
```
Comment (hoorayimhelping) [inline ui/src/cloud/apis/demodata.ts:38]: can you add a // todo here to use `fetch` since i think that's the direction we're going?

File: ui/src/cloud/apis/demodata.ts [hunk 5]
```diff
@@ -106,3 +99,30 @@ export const fetchDemoDataBuckets = async (): Promise<Bucket[]> => {
     return [] // demodata bucket fetching errors should not effect regular bucket fetching
   }
 }
+
+export const getNormalizedDemoDataBucket = async (
+  bucketID: string
+): Promise<NormalizedSchema<BucketEntities, string>> => {
+  const resp = await getBucket({bucketID})
```
Comment (hoorayimhelping) [inline ui/src/cloud/apis/demodata.ts:106]: just a sanity check: we don't need to wrap this in a `try / catch` do we ?

File: ui/src/shared/components/notifications/Notifications.tsx [hunk 6]
```diff
@@ -1,53 +1,99 @@
 import React, {PureComponent} from 'react'
+import {Link} from 'react-router'
 import {connect} from 'react-redux'
-import {Notification as NotificationType} from 'src/types/notifications'
-import Notification from 'src/shared/components/notifications/Notification'
+import {get} from 'lodash'
 
-interface Props {
+//Actions
```
Comment (hoorayimhelping) [inline ui/src/shared/components/notifications/Notifications.tsx:6]: *cough* this seems like a great thing to split up into a separate PR that can get merged ahead of this one.

File: ui/src/cloud/actions/demodata.ts [hunk 7]
```diff
@@ -60,85 +64,72 @@ export const getDemoDataBuckets = () => async (
   try {
     const buckets = await getDemoDataBucketsAJAX()
 
-    dispatch(setDemoDataStatus(RemoteDataState.Done))
     dispatch(setDemoDataBuckets(buckets))
   } catch (error) {
     console.error(error)
```
[the remark this answers is not in the document]
Comment (ebb-tide) [reply ui/src/cloud/actions/demodata.ts:69]: I agree. I've removed these for console.errors when we also notify for the user.

File: ui/src/cloud/apis/demodata.ts (hunk 5 above)
Comment (ebb-tide) [reply ui/src/cloud/apis/demodata.ts:106]: it's called in a function that try/catch'es, and I want that function to handle the error conditions. I would like to write a try/catch which throws the error up, but apparently that is against our linter rules. It does make me worry that someone will use this function without properly try/catching but... yeah.

File: ui/src/shared/components/notifications/Notifications.tsx (hunk 6 above)
Comment (ebb-tide) [reply ui/src/shared/components/notifications/Notifications.tsx:6]: totally agree and thank you so much. I definitely tried cherry picking apart, but decided to just go for it in this case 😬
